### ISSUE TITLE
[feature]requesting for pretrained weights

### ISSUE THREAD

**Issue by jianyin2016** [prabhuomkar/pytorch-cpp#51]
**Is your feature request related to a problem? Please describe.**

>    I would like to train a CNN-classifier with my custom data using the widely-used models like ResNet series，I found it is useful to initialize the model weights with ImageNet pretrained weights, and it is easy to implement with the `torch::load` API when the image channels of my dataset is 3, the same as ImageNet，under which situation no change should be made to the `Conv1` layer.
It is the other situation when I try to train with gray images，as the `Conv1` weights is supposed to be of  `in_channels=3`,  In the python fronten, I guess this maybe solved but imdieatly repalce the model.conv1 like this:
~~~
model.conv1 = nn.Conv2d(in_channels=1, out_channels=64, kernel_size=7, stride=2, padding=3, bias=False)
~~~
but as for the C++ fronten, repalcing seems not work:
~~~
model->conv1 = torch::nn::Conv2d(torch::nn::Conv2dOptions(in_channels, 64, 7).stride(2).padding(3).bias(false).dilation(1));
~~~

**Describe the solution you'd like**
> Correcting the API use to rightly loading the pretrained weights. 

**Describe alternatives you've considered**
> Maybe a pretrained model on gray image dataset would bypass the problem.

**Additional context**
> Exception occurs during the model forward process.

**Comment by mfl28** [comment]
Hi @jianyin2016 !
I think the easiest way to achieve what you want to do - if I understand correctly - is:
1. Create your model as you want it in python:

```Python
model = torchvision.models.resnet...(pretrained=True)
model.conv1 = nn.Conv2d(in_channels=1, out_channels=64, kernel_size=7, stride=2, padding=3, bias=False)
```
2. Save it as a scriptmodule:
```Python
example = torch.rand(1, 1, 224, 224)
traced_script_module = torch.jit.trace(model, example)
traced_script_module.save("my_resnet.pt")
```
3. Load it in C++ via `auto model = torch::jit::load("path/to/my_resnet.pt")`

There is a nice and much more detailed description of this process in the official Pytorch tutorials: 
[Loading a TorchScript Model in C++](https://pytorch.org/tutorials/advanced/cpp_export.html)

**Comment by jianyin2016** [comment]
Hi，@mfl28 , thanks for your nice advice.

There is no doubt that your solution will work, I quite believe in that , but the solution seems to be sort of bypassing rather than solving. I think maybe it is not the right way using the C++ API while at the same time relying too much on the PYTHON parts. so, still, I am here asking for a solution absolutly within libtorch.

thanks!

**Comment by mfl28** [comment]
To my knowledge it is currently not possible to do this completely within C++, as there is no `load_state_dict()` function for models in libtorch (see  [this recent issue](https://github.com/pytorch/pytorch/issues/36577) in the official pytorch repo) to load pickled weight files. Maybe you can find more information on this in the [Pytorch forum](https://discuss.pytorch.org/) or the official repo's issues.
